### PR TITLE
CSL-3133 Tracking space trimming

### PR DESCRIPTION
**Pull request by sblaurock**
Related to #183 

Do not trim space from "term" related tracking parameters (`term`, `original_query`, `search_term`)

### CODE REVIEW

**Comment by esezen** [comment]
@sblaurock from what I understand, `trimNonBreakingSpaces` replaces non-breaking spaces with normal spaces and trims from both ends. I understand we want to get rid of the trimming, are we also getting rid of the replace part or should we split that function into two parts and apply one or the other in different scenarios?

**Comment by sblaurock** [comment]
> @sblaurock from what I understand, `trimNonBreakingSpaces` replaces non-breaking spaces with normal spaces and trims from both ends. I understand we want to get rid of the trimming, are we also getting rid of the replace part or should we split that function into two parts and apply one or the other in different scenarios?

Great thought, Enes. I'll adjust - I think your suggestion is the better way to go. Thank you!

**Comment by sblaurock** [comment]
@esezen mind taking another look when you find a moment? thank you!

File: src/modules/tracker.js [hunk 1]
```diff
@@ -516,7 +516,7 @@ class Tracker {
     if (term && typeof term === 'string') {
       // Ensure parameters are provided (required)
       if (parameters && typeof parameters === 'object' && !Array.isArray(parameters)) {
-        const url = `${this.options.serviceUrl}/autocomplete/${helpers.encodeURIComponentRFC3986(helpers.trimNonBreakingSpaces(term))}/search?`;
```
Comment (esezen) [inline src/modules/tracker.js:519]: Do we want to call `normalizeSpaces` here? (Without the trim)

Comment (sblaurock) [reply src/modules/tracker.js:519]: excellent catch once again

**Review by esezen — APPROVED** [review]
LGTM!

**Pull request merged**